metrics/sources/kubelet/kubelet.go

```diff
@@ -132,25 +157,12 @@ func (this *kubeletMetricsSource) decodeMetrics(c *cadvisor.ContainerInfo) (stri
 			}
 		}
 
+		// No Kubernetes metadata so treat this as a system container.
 		if cName == "" || ns == "" || podName == "" {
```

Comment:
Yes, but at that moment we don't know to which pod the container belongs. Ideally we would take container id and look up the correct pod/container, but here we don't use this mechanisms (for simplicity, maybe we should) and we need a name. In Kubernetes 1.0.x the name was NOT available as an annotation so we extract it from docker-specific container name, what can possibly (but unlikely) fail.